docs/develop/java/spring/rate-limiting/fixed-window/index-fixed-window-reactive-gears.mdx

```diff
@@ -0,0 +1,290 @@
+the `key`, the quota and the TTL seconds in the future.
+
+As we've have done previously, if the function returns `false` we let the request through, otherwise
+we return an `HTTP 429`:
+
+```java
+@Override
+public Mono<ServerResponse> filter(ServerRequest request, HandlerFunction<ServerResponse> next) {
+  int currentMinute = LocalTime.now().getMinute();
+  String key = String.format("rl_%s:%s", requestAddress(request.remoteAddress()), currentMinute);
+
+  RedisGearsCommands<String, String> gears = connection.sync();
+
+  List<Object> results = gears.trigger("RateLimiter", key, Long.toString(maxRequestPerMinute), "59");
+  if (!results.isEmpty() && !Boolean.parseBoolean((String) results.get(0))) {
+    return next.handle(request);
+  } else {
+    return ServerResponse.status(TOO_MANY_REQUESTS).build();
+  }
+}
+```
+
+## Testing with curl
+
+Once again, we use curl loop to test the limiter:
+
+```
+for n in {1..22}; do echo $(curl -s -w " :: HTTP %{http_code}, %{size_download} bytes, %{time_total} s" -X GET http://localhost:8080/api/ping); sleep 0.5; done
+```
+
+You should see the 21st request being rejected:
+
+```bash
+➜ for n in {1..22}; do echo $(curl -s -w " :: HTTP %{http_code}, %{size_download} bytes, %{time_total} s" -X GET http://localhost:8080/api/ping); sleep 0.5; done
+PONG :: HTTP 200, 4 bytes, 0.064786 s
+PONG :: HTTP 200, 4 bytes, 0.009926 s
+PONG :: HTTP 200, 4 bytes, 0.009546 s
+PONG :: HTTP 200, 4 bytes, 0.010189 s
+PONG :: HTTP 200, 4 bytes, 0.009399 s
+PONG :: HTTP 200, 4 bytes, 0.009210 s
+PONG :: HTTP 200, 4 bytes, 0.008333 s
+PONG :: HTTP 200, 4 bytes, 0.008009 s
+PONG :: HTTP 200, 4 bytes, 0.008919 s
+PONG :: HTTP 200, 4 bytes, 0.009271 s
+PONG :: HTTP 200, 4 bytes, 0.007515 s
+PONG :: HTTP 200, 4 bytes, 0.007057 s
+PONG :: HTTP 200, 4 bytes, 0.008373 s
+PONG :: HTTP 200, 4 bytes, 0.007573 s
+PONG :: HTTP 200, 4 bytes, 0.008209 s
+PONG :: HTTP 200, 4 bytes, 0.009080 s
+PONG :: HTTP 200, 4 bytes, 0.007595 s
+PONG :: HTTP 200, 4 bytes, 0.007955 s
+PONG :: HTTP 200, 4 bytes, 0.007693 s
+PONG :: HTTP 200, 4 bytes, 0.008743 s
+:: HTTP 429, 0 bytes, 0.007226 s
+:: HTTP 429, 0 bytes, 0.007388 s
+```
+
+If we run Redis in monitor mode, we should see the Lua calls to `RG.TRIGGER` and under that you should see the
+calls to `GET`, `INCR` and `EXPIRE` for allowed requests:
+
+```bash
+1631249244.006212 [0 172.17.0.1:56036] "RG.TRIGGER" "RateLimiter" "rl_localhost:47" "20" "59"
+1631249244.006995 [0 ?:0] "GET" "rl_localhost:47"
+1631249244.007182 [0 ?:0] "INCR" "rl_localhost:47"
+1631249244.007269 [0 ?:0] "EXPIRE" "rl_localhost:47" "59"
+```
+
+And for rate limited request you should see only the call to `GET`:
+
+```
+1631249244.538478 [0 172.17.0.1:56036] "RG.TRIGGER" "RateLimiter" "rl_localhost:47" "20" "59"
+1631249244.538809 [0 ?:0] "GET" "rl_localhost:47"
+```
+
+The complete code for this implementation is under the branch [`with_gears`](https://github.com/redis-developer/fixed-window-rate-limiter/tree/with_gears).
```

Comment:
Maybe put another header above this for Resources, along with a link to RedisGears' OSS page